drivers/SmartThings/zwave-thermostat/src/aeotec-radiator-thermostat/init.lua

```diff
@@ -97,6 +99,52 @@ local function device_added(self, device)
   do_refresh(self, device)
 end
 
+--TODO: Update this once we've decided how to handle setpoint commands
+local function convert_to_device_temp(command_temp, device_scale)
+  -- under 40, assume celsius
+  if (command_temp < 40 and device_scale == ThermostatSetpoint.scale.FAHRENHEIT) then
+    command_temp = utils.c_to_f(command_temp)
+  elseif (command_temp >= 40 and (device_scale == ThermostatSetpoint.scale.CELSIUS or device_scale == nil)) then
+    command_temp = utils.f_to_c(command_temp)
+  end
+  return command_temp
+end
+
+local function set_setpoint_factory(setpoint_type)
+  return function(driver, device, command)
+    local scale = device:get_field(constants.TEMPERATURE_SCALE)
+    local value = convert_to_device_temp(command.args.setpoint, scale)
+
+    local set
+    -- This device expects to get fractional values as a uint16 with
+    -- a provided "precision" of 1 to indicate that the integer should
+    -- be divided by 10. i.e. the value of 21.5 is 215 with precision 1
+    if value % 1 == 0.5 then
+      set = ThermostatSetpoint:Set({
+        setpoint_type = setpoint_type,
+        scale = scale,
+        value = value,
+        precision = 1,
+        size = 2
+      })
```

Comment:
I'd honestly include this in the base driver's method, since this seems like an error we'd want to address everywhere